CardScan/Classes/Ocr.swift

```diff
@@ -110,23 +110,24 @@ public class Ocr {
         if findFour.cardDetected ?? false {
             self.scanStats.cardsDetected += 1
         }
-        
-        if let number = number {
+        if let squareCardImage = squareCardImage, let fullCardImage = fullCardImage {
+            let croppedCardSize = CGSize(width: croppedCardImage.width, height: croppedCardImage.height)
             // Note: the onScanComplete method is called in ScanBaseViewController
             let xmin = findFour.predictedBoxes.map { $0.minX }.min() ?? 0.0
             let xmax = findFour.predictedBoxes.map { $0.maxX }.max() ?? 0.0
             let ymin = findFour.predictedBoxes.map { $0.minY }.min() ?? 0.0
             let ymax = findFour.predictedBoxes.map { $0.maxY }.max() ?? 0.0
             let numberBoundingBox = CGRect(x: xmin, y: ymin, width: (xmax - xmin), height: (ymax - ymin))
             
-            if let squareCardImage = squareCardImage, let fullCardImage = fullCardImage {
-                let croppedCardSize = CGSize(width: croppedCardImage.width, height: croppedCardImage.height)
+             if let number = number {
                 self.scanEventsDelegate?.onNumberRecognized(number: number, expiry: findFour.expiry, numberBoundingBox: numberBoundingBox, expiryBoundingBox: findFour.expiryBoxes.first, croppedCardSize: croppedCardSize, squareCardImage: squareCardImage, fullCardImage: fullCardImage)
+            
+                self.scanStats.algorithm = findFour.algorithm
+                self.updateStats(model: findFour.modelString, boxes: findFour.predictedBoxes, image: croppedCardImage, number: number, cvvBoxes: findFour.cvvBoxes)
+                return number
             }
             
-            self.scanStats.algorithm = findFour.algorithm
-            self.updateStats(model: findFour.modelString, boxes: findFour.predictedBoxes, image: croppedCardImage, number: number, cvvBoxes: findFour.cvvBoxes)
-            return number
+            self.scanEventsDelegate?.onFrameDetected(number: number, expiry: findFour.expiry, numberBoundingBox: numberBoundingBox, expiryBoundingBox: findFour.expiryBoxes.first, croppedCardSize: croppedCardSize, squareCardImage: squareCardImage, fullCardImage: fullCardImage)
```

Comment:
I would have assumed that this would go in the else side of things. So in other words we either detect a number or a frame but never both.